Sources/MapboxDirections/RouteRoadClassesViolations.swift

```diff
@@ -0,0 +1,85 @@
+
+import Foundation
+
+/**
+ Description of `RoadClasses` that were meant to be avoided during routing but still got used in a `Route`.
+ 
+ If something in `RouteOptions.roadClassesToAvoid` prevents the origin or destination from connecting to the main road network, the Directions API will temporarily ignore the road class constraints in order to reach the main road network.
+ */
+public struct RouteRoadClassesViolations {
+    /**
+     Related `Route` object
+     */
+    public let route: Route
+    
+    /**
+     List of all existing violations, down to `RouteStep` granularity.
+     */
+    public let violations: [RoadClassExclusionViolation]
+    
+    /**
+     Filters `violations` to search for specific leg and step.
+     
+     - parameter legIndex: Index of a leg inside related `Route`to search in.
+     - parameter stepIndex: Index of a step inside given `Route`'s leg.
+     - returns: Array of `RoadClassExclusionViolation` under given indicies.
+     
+     Passing `nil` as `stepIndex` will result in searching for all steps.
+     */
+    public func violations(at legIndex: Int, stepIndex: Int? = nil) -> [RoadClassExclusionViolation] {
+        return fetchViolations(at: legIndex, stepIndex: stepIndex, intersectionIndex: nil)
+    }
+    
+    /**
+     Filters `violations` to search for specific leg, step and intersection.
+     
+     - parameter legIndex: Index of a leg inside related `Route`to search in.
+     - parameter stepIndex: Index of a step inside given `Route`'s leg.
+     - parameter intersectionIndex: Index of an intersection inside given `Route`'s leg and step.
+     - returns: Array of `RoadClassExclusionViolation` under given indicies.
+     
+     Passing `nil` as `intersectionIndex` will result in searching for all intersections of given step.
+     */
+    public func violations(at legIndex: Int, stepIndex: Int, intersectionIndex: Int?) -> [RoadClassExclusionViolation] {
```

Comment:
Remove the `at` label in favor of the default `legIndex`: there doesn’t seem to be anything about the leg index that qualifies for “at” more than the step and intersection indices do.